docs/capabilities/event-emitter.md

```diff
@@ -0,0 +1,99 @@
+# Event Emitter
+
+The Event Emitter is a capability provided by AlgoKit Utils that allows for asynchronous event handling. It provides a flexible mechanism for emitting and listening to custom events, which can be particularly useful for debugging and extending functionality not available in the `algokit-utils-ts` package.
+
+## `AsyncEventEmitter`
+
+The [`AsyncEventEmitter`](../code/classes/types_async_event_emitter.AsyncEventEmitter.md) is a class that manages asynchronous event emission and subscription.
+
+To use the `AsyncEventEmitter`, you can import it directly:
+
+```typescript
+import { AsyncEventEmitter } from '@algorandfoundation/algokit-utils/types/async-event-emitter'
+
+const emitter = new AsyncEventEmitter()
+```
+
+## Event Types
+
+The `EventType` enum defines the built-in event types:
+
+```typescript
+enum EventType {
+  TxnGroupSimulated = 'TxnGroupSimulated',
+  AppCompiled = 'AppCompiled',
+}
+```
+
+These events are primarily used for debugging purposes and are handled by the [`algokit-utils-ts-debug`](https://github.com/algorandfoundation/algokit-utils-ts-debug) package. However, the `AsyncEventEmitter` is designed to be generic and can be extended to handle custom event types as well.
+
+## Emitting Events
```

Comment:
The emitting side is mostly an internal concern for the utils library, so I'd remove this section and wouldn't focus on that side of things.